lib/mix/lib/mix.ex

```diff
@@ -428,15 +430,25 @@ defmodule Mix do
     Mix.State.put(:debug, debug)
   end
 
+  @doc """
+  Raises a Mix error that is nicely formatted, defaulting to exit code `1`.
+  """
+  @spec raise(binary) :: no_return
```

Comment:
```suggestion
  @doc since: "1.0"
  @spec raise(binary) :: no_return
```

Reply:
Is there a need for this? Do we do this for older functions? 🤔

Reply:
To be honest I don't know why it is not specified in every function. But no function introduced in v1.0 is annotated

Reply:
We can assume that if a function is not annotated it has been around since 1.0.

Reply:
I believe we don't annotate anything before v1.4. There is no point in annotating those because v1.4 is not even supported anymore, so people should really migrate.